### ISSUE TITLE
Add in UCAC/URAT proper motions where Gaia is missing

### ISSUE THREAD

**Issue by geordie666** [desihub/desitarget#520]
In 1.5% of the sky we have no proper motions from Gaia for our "typical" GFA objects. For GFA targets with missing Gaia proper motions, supplement with [the URAT catalog](http://cdsarc.u-strasbg.fr/viz-bin/Cat?I/329).

**Comment by djschlegel** [comment]
Has anyone used this URAT catalog yet?  The only place I can seem to find the files are in the v12 directory of http://cdsarc.u-strasbg.fr/ftp/I/329/URAT1/ , but it looks like that's a straight binary file (not FITS binary, just raw binary).  Do-able, but really a pain.  And who knows, maybe they made these files on a VAX with a mix of big-endian and little-endian... :<

**Comment by sbailey** [comment]
I haven't successfully parsed these files, but a bit of other info:

data model: http://cdsarc.u-strasbg.fr/ftp/I/329/ReadMe

directory with fortran code for reading them and 2 files extracted into ascii to cross check: http://cdsarc.u-strasbg.fr/ftp/I/329/URAT1/access/

astropy code that knows how to parse the ReadMe, but apparently is intended for reading the ascii version of the output not raw binary: https://docs.astropy.org/en/stable/api/astropy.io.ascii.Cds.html

Ugh.

**Comment by geordie666** [comment]
I modified the provided fortran code to successfully parse the binary files, and verified that I could recreate the 2 files extracted into ascii.

@sbailey: Hopefully there will be no licensing problem, or DESI software standard, to prevent me committing my version of the fortran to git, when I'm done incorporating URAT into desitarget.gfa.

**Comment by ameisner** [comment]
What about HSOY?

https://arxiv.org/pdf/1701.02629.pdf

It's a Gaia DR1 + PPMXL hybrid. That's what I used for my unWISE astrometric recalibrations before Gaia DR2 was available.

-Aaron

**Comment by sbailey** [comment]
> I modified the provided fortran code to successfully parse the binary files, and verified that I could recreate the 2 files extracted into ascii.
> 
> @sbailey: Hopefully there will be no licensing problem, or DESI software standard, to prevent me committing my version of the fortran to git, when I'm done incorporating URAT into desitarget.gfa.

They don't provide an explicit license, but as long as you very clearly credit them with the original code, and note that you had to modify it and where the original is, I think we're fine.

However: let's do a single reformatting of their files into FITS files and keep those under subdirectories of cosmo/data/ or desi/external/ and then have desitarget use that, and *not* run fortran code on-the-fly to do conversions every time.  It's still good practice to cache the code that did the conversion though.

I have no opinion about HSOY vs. UCAC/URAT, other than it has a slightly better sounding acronym.

**Comment by ameisner** [comment]
I investigated HSOY a bit more, and found that it has "missing" regions largely (although not exactly) degenerate with those of the regions lacking Gaia DR2 proper motions. So I don't really think HSOY is going to provide a solution here. I never noticed the problematic stripes in either HSOY or Gaia DR2 because I'd always been working with 1.5 deg x 1.5 deg unWISE tiles that are much larger than the GFA footprints.

**Comment by geordie666** [comment]
I'm finding that about 90% of the UCAC/URAT sources have proper motions within +/-20 mas per year (0.02"/yr) of their Gaia DR2 counterpart. About 80% have proper motions within +/-10 mas per year (0.01"/yr) of Gaia.

@ameisner: You know more about astrometry than me. Do you think these numbers sound reasonable? If you'd prefer HSOY, do you happen to have it downloaded somewhere at NERSC so I don't have to wait on the overnight wget?

Unless we think UCAC/URAT isn't good enough for our purposes (the very few GFA cameras that won't cover a Gaia source), then I'll keep pushing through with UCAC/URAT, as I've already done substantial work scraping and reformatting it.

**Comment by geordie666** [comment]
@ameisner: I was writing my comment as you posted yours! I guess I'll push on with UCAC/URAT.

**Comment by geordie666** [comment]
Addressed in #526.
